src/cli/pytest_commands/consume.py

```diff
@@ -6,64 +6,51 @@
 
 import click
 
-from .base import ArgumentProcessor, PytestCommand, PytestExecution, common_pytest_options
+from .base import ArgumentProcessor, PytestCommand, common_pytest_options
 from .processors import ConsumeCommandProcessor, HelpFlagsProcessor, HiveEnvironmentProcessor
 
 
-class ConsumeCommand(PytestCommand):
-    """Pytest command for consume operations."""
-
-    def __init__(self, command_paths: List[Path], is_hive: bool = False, command_name: str = ""):
-        """Initialize consume command with paths and processors."""
-        processors: List[ArgumentProcessor] = [HelpFlagsProcessor("consume")]
-
-        if is_hive:
-            processors.extend(
-                [
-                    HiveEnvironmentProcessor(command_name=command_name),
-                    ConsumeCommandProcessor(is_hive=True),
-                ]
-            )
-        else:
-            processors.append(ConsumeCommandProcessor(is_hive=False))
-
-        super().__init__(
-            config_file="pytest-consume.ini",
-            argument_processors=processors,
+def create_consume_command(
+    *,
+    test_paths: List[Path],
+    is_hive: bool = False,
+    command_name: str = "",
+) -> PytestCommand:
+    """Initialize consume command with paths and processors."""
+    processors: List[ArgumentProcessor] = [HelpFlagsProcessor("consume")]
+
+    if is_hive:
+        processors.extend(
+            [
+                HiveEnvironmentProcessor(command_name=command_name),
+                ConsumeCommandProcessor(is_hive=True),
+            ]
         )
-        self.command_paths = command_paths
-
-    def create_executions(self, pytest_args: List[str]) -> List[PytestExecution]:
-        """Create execution with test paths prepended."""
-        processed_args = self.process_arguments(pytest_args)
+    else:
+        processors.append(ConsumeCommandProcessor(is_hive=False))
 
-        # Prepend test paths to arguments
-        test_path_args = [str(p) for p in self.command_paths]
-        final_args = test_path_args + processed_args
-
-        return [
-            PytestExecution(
-                config_file=self.config_file,
-                args=final_args,
-            )
-        ]
+    return PytestCommand(
+        config_file="pytest-consume.ini",
+        argument_processors=processors,
+        test_paths=test_paths,
+    )
 
 
 def get_command_paths(command_name: str, is_hive: bool) -> List[Path]:
```

Comment:
Thanks for the better variable naming! How about...?
```suggestion
def get_test_paths(command_name: str, is_hive: bool) -> List[Path]:
```

Reply:
Used `get_static_test_paths`.